src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -0,0 +1,144 @@
+    intl: IntlShape,
+    metadataColumnsToShow: Array<string>,
+} & InjectIntlProvidedProps;
+
+type CellRendererArgs = {
+    columnIndex: number,
+    key: string,
+    rowIndex: number,
+    style: StringAnyMap,
+};
+
+class MetadataBasedItemList extends React.Component<Props> {
+    props: Props;
+
+    getColumnWidth(width: number): Function {
+        const { metadataColumnsToShow }: Props = this.props;
+
+        return ({ index }: { index: number }): number => {
+            if (index === 0) {
+                return FILE_ICON_COLUMN_WIDTH;
+            }
+
+            if (index === 1) {
+                return FILENAME_COLUMN_WIDTH;
+            }
+
+            const availableWidth = width - FILENAME_COLUMN_WIDTH - FILE_ICON_COLUMN_WIDTH; // total width minus width of sticky columns
+            // Maintain min column width, else occupy the rest of the space equally
+            return Math.max(availableWidth / metadataColumnsToShow.length, MIN_METADATA_COLUMN_WIDTH);
+        };
+    }
+
+    getGridCell(columnIndex: number, rowIndex: number) {
+        const { currentCollection, metadataColumnsToShow }: Props = this.props;
+        const { items } = currentCollection;
+        const { name, metadata } = items[rowIndex];
+
+        if (columnIndex === 0) {
+            return <FileIcon dimension={FILE_ICON_SIZE} extension={getFileExtension(name)} />;
+        }
+
+        if (columnIndex === 1) {
+            return <span>{name}</span>;
+        }
+
+        const { data = {} } = metadata;
+        const mdFieldName = metadataColumnsToShow[columnIndex - FIXED_COLUMNS_NUMBER];
+        return <span>{data[mdFieldName]}</span>;
+    }
+
+    getGridHeader(columnIndex: number, style: StringAnyMap) {
+        const { intl, metadataColumnsToShow } = this.props;
+
+        let displayName;
+
+        switch (columnIndex) {
+            case 0:
+                displayName = null; // No column header for file icon
+                break;
+            case 1:
+                displayName = <span>{intl.formatMessage(messages.name)}</span>; // Column header "Name" for fileName
+                break;
+            default:
+                displayName = <span>{metadataColumnsToShow[columnIndex - FIXED_COLUMNS_NUMBER]}</span>; // Column to show
+        }
+
+        return (
+            <div className="bdl-MetadataBasedItemList-gridHeader" style={style}>
+                {displayName}
+            </div>
+        );
+    }
+
+    cellRenderer = ({ columnIndex, key, rowIndex, style }: CellRendererArgs) => {
+        if (rowIndex === 0) {
+            // Render Header Row
+            return this.getGridHeader(columnIndex, style);
+        }
+
+        const classes = classNames('bdl-MetadataBasedItemList-gridCell', {
+            'bdl-MetadataBasedItemList-filename': columnIndex === 1, // file name cell
+        });
+        // Render data row
+        const cellData = this.getGridCell(columnIndex, rowIndex - 1);
```

Comment:
Minor, but this can be inlined in the JSX, since the variable is only used once.